framework/mesh/MeshGenerator/SplitFileMeshGenerator.h

```diff
@@ -0,0 +1,56 @@
+#ifndef CHITECH_SPLITFILEMESHGENERATOR_H
+#define CHITECH_SPLITFILEMESHGENERATOR_H
+
+#include "MeshGenerator.h"
+
+namespace chi_data_types
+{
+class ByteArray;
+}
+
+namespace chi_mesh
+{
+
+/**Generates the mesh only on location 0, thereafter partitions the mesh
+ * but instead of broadcasting the mesh to other locations it creates binary
+ * mesh files for each location.*/
+class SplitFileMeshGenerator : public MeshGenerator
+{
+public:
+  static chi::InputParameters GetInputParameters();
+  explicit SplitFileMeshGenerator(const chi::InputParameters& params);
+
+  void Execute() override;
+
+protected:
+  void WriteSplitMesh(const std::vector<int64_t>& cell_pids,
+                      const UnpartitionedMesh& umesh,
+                      int num_parts);
+  static void SerializeCell(const UnpartitionedMesh::LightWeightCell& cell,
+                            chi_data_types::ByteArray& serial_buffer);
+  typedef std::pair<int, uint64_t> CellPIDGID;
+  struct SplitMeshInfo
```

Comment:
What about `LocalMeshInfo`?

Reply:
Negative because the cells and the vertices in the mesh is not exclusively local. They may have ghost relations

Reply:
Hmmm. Since this is a `SplitFileMesh` saying `SplitMeshInfo` sounds like it is a global thing, as opposed to a struct containing the local portion (plus ghosts) of the global mesh. Within the discretizations, we can get a ghost DoF from `MapDOFLocal`. While the ghost DoF is not truly local, there is local access. I would make a similar argument here. A local mesh is all of the information needed locally to fit together with the global mesh. Take this for what you will